jax/numpy/lax_numpy.py

```diff
@@ -3088,6 +3088,25 @@ def cov(m, y=None, rowvar=True, bias=False, ddof=None, fweights=None,
   X_T = X.T if w is None else (X * w).T
   return true_divide(dot(X, X_T.conj()), f).squeeze()
 
+@_wraps(onp.convolve)
+def convolve(x, y, mode='full'):
+  if len(shape(x)) != 1 or len(shape(y)) != 1:
+    msg = ("jax.numpy.convolve only accept 1d array")
+    raise ValueError(msg)
+  if shape(y)[0] > shape(x)[0]:
+    x, y = y, x
+  x, y = _promote_dtypes(x, y)
+  x_jax = lax.reshape(x, (1, 1, len(x)))
+  y_jax = flip(lax.reshape(y, (1, 1, len(y))))
+  if mode.lower() == 'full':
+    return ravel(lax.conv_general_dilated(x_jax, y_jax, [1], [(len(y)-1, len(y)-1)]))
+  elif mode.lower() == 'valid':
+    return ravel(lax.conv(x_jax, y_jax, (1,), "VALID"))
+  elif mode.lower() == 'same':
+    pads = lax.padtype_to_pads(x_jax.shape[2:], y_jax.shape[2:], (1,), "SAME")
```

Comment:
Could we express this more directly? I think calling `padtype_to_pads` is actually just making the code more complicated, not less. Here I think you just want:
```
low = (len(y) + 1) // 2
high = len(y) - low
```

(although you'd need to double check that, I didn't test that.)